quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java

```diff
@@ -0,0 +1,103 @@
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @Autowired
+    private UserDao userDao;
+
+    /** Implementation of GetAllQuestions endpoint.
+     * This method expose end point /question/all
+     *
+     * @param authorization : authorization token.
+     * @return List<QuestionDetailsResponse>
+     * @throws AuthorizationFailedException
+     */
+    @RequestMapping(method = RequestMethod.GET, path = "/question/all", produces = MediaType.APPLICATION_JSON_UTF8_VALUE )
+    public ResponseEntity<List<QuestionDetailsResponse>> getAllQuestions(@RequestHeader("authorization") final String accessToken) throws AuthorizationFailedException {
+
+        UserAuthTokenEntity authTokenEntity =  authenticationService.getUserByToken(accessToken);
+        if (authTokenEntity != null) {
+
+            List<QuestionEntity> allQuestions = questionsService.getAllQuestions(accessToken);
+
+            return getListResponseEntity(allQuestions);
+        }
+        else
+        {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+    }
+
+    /** Handled endpoint to display all questions. Prepares the List of All question to be displayed.
+     * This method expose end point /question/all
+     *
+     * @param allQuestion : List of QuestionEntity
+     * @return ResponseEntity
+     *
+     */
+    private ResponseEntity<List<QuestionDetailsResponse>> getListResponseEntity(List<QuestionEntity> allQuestions) {
+        QuestionEntity questionEntity;
+        if (allQuestions.size() != 0) {
+            List<QuestionDetailsResponse> displayAllQuestions = new ArrayList<>();
+            for (int i = 0; i < allQuestions.size(); i++) {
+                questionEntity = allQuestions.get(i);
+                QuestionDetailsResponse questionDetailsResponse = new QuestionDetailsResponse().id(questionEntity.getUuid()).content(questionEntity.getContent());
+                displayAllQuestions.add(questionDetailsResponse);
+            }
+            return new ResponseEntity<List<QuestionDetailsResponse>>(displayAllQuestions, HttpStatus.OK);
+        }
+        else {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+    }
+
+    /** Implementation of GetAllQuestions endpoint.
+     * This method expose end point /question/all
+     *
+     * @param authorization : authorization token.
+     * @return List<QuestionDetailsResponse>
+     * @throws AuthorizationFailedException
+     */
+    @RequestMapping(method = RequestMethod.GET, path = "/question/all/{userId}", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<List<QuestionDetailsResponse>> getAllQuestionByUser(@RequestHeader("authorization") final String accessToken, @PathVariable("userId") final String userId) throws AuthorizationFailedException, UserNotFoundException {
+
+        UserAuthTokenEntity authTokenEntity =  authenticationService.getUserByToken(accessToken);
```

Comment:
getUserByToken logic should be inside business handling code

Reply:
you are already checking authenticationService.getUserByToken(accessToken);  in questionsService.getAllQuestionsByUser(userId, accessToken)

Reply:
Modified the Code.